From a7289eeff4ac9e3ca2d2289672c923518433124c Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 8 Dec 2021 09:26:58 -0800
Subject: [PATCH] Musl/linux can not use RTLD_DEEPBIND

Exclude non-glibc linux systems along with android
Fixes
src/tbb/dynamic_link.cpp:417:29: error: use
 of undeclared identifier 'RTLD_DEEPBIND'                                                                                    |             flags = flags | RTLD_DEEPBIND;
|                             ^

Upstream-Status: Submitted [https://github.com/oneapi-src/oneTBB/pull/684]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/tbb/dynamic_link.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tbb/dynamic_link.cpp b/src/tbb/dynamic_link.cpp
index 12f35cfd..942c1c64 100644
--- a/src/tbb/dynamic_link.cpp
+++ b/src/tbb/dynamic_link.cpp
@@ -413,7 +413,7 @@ namespace r1 {
         int flags = RTLD_NOW;
         if (local_binding) {
             flags = flags | RTLD_LOCAL;
-#if __linux__ && !__ANDROID__ && !__TBB_USE_SANITIZERS
+#if (__linux__ && __GLIBC__) && !__TBB_USE_SANITIZERS
             flags = flags | RTLD_DEEPBIND;
 #endif
         } else {
-- 
2.34.1

